### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**

### CODE REVIEW

**Review by Yekatery — APPROVED** [review]
Well done!

**Review by Andr1yK — CHANGES_REQUESTED** [review]
check by checklist :)

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,22 @@
...
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete state[keys];
+        }
+        break;
 
+      case 'clear':
+        Object.keys(state).forEach(k => delete state[k]);
```
Comment (Andr1yK) [inline src/transformState.js:21]:
```suggestion
        Object.keys(state).forEach(key => delete state[key]);
```

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,22 @@
...
-  // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
```
Comment (Andr1yK) [inline src/transformState.js:15]:
```suggestion
        for (const key of action.keysToRemove) {
```

**Review by Andr1yK — APPROVED** [review]

**Pull request closed without merging**